File: apps/accounts/models/term.py
from django.db import models
from common.models import BaseModel

class Term(BaseModel):
    approved_by = models.CharField(max_length=255, blank=True, null=True)
    day_cut_off_due = models.IntegerField(blank=True, null=True)
    day_cut_off_invoice = models.IntegerField(blank=True, null=True)
    days_discount = models.IntegerField(blank=True, null=True)
    days_due = models.IntegerField(blank=True, null=True)
    days_in_period = models.IntegerField(blank=True, null=True)
    description = models.CharField(max_length=255, blank=True, null=True)
    discount_rate = models.FloatField(blank=True, null=True)
    period_count = models.IntegerField(blank=True, null=True)

    def __str__(self):
        return f"Term ({self.id})"

    class Meta:
        db_table = 'terms'
    

File: src/api/apps/accounts/models/term/Term.ts
// WC3 ↔ R25 TypeScript alignment interface for Term
export interface Term {
  id: number;
  uuid?: string;
  ida?: string;
  dt_created?: number;
  dt_modified?: number;
  version?: number;
  is_active?: boolean;
  is_deleted?: boolean;
  is_archived?: boolean;
  name?: string;
  approved_by?: string;
  day_cut_off_due?: number;
  day_cut_off_invoice?: number;
  days_discount?: number;
  days_due?: number;
  days_in_period?: number;
  description?: string;
  discount_rate?: number;
  period_count?: number;
  metadata?: Record<string, any>;
  refs?: Record<string, any>;
  prefs?: Record<string, any>;
  comments?: Record<string, any>;
  health_rating?: number;
}
